### PR TITLE
Add components check list

### PR DESCRIPTION
**Pull request by eerhardt**

### CODE REVIEW

File: src/Components/README.md [hunk 1]
```diff
@@ -163,3 +163,22 @@ builder.AddAzureServiceBus(settings =>
 ```
 
 - If both secret and passwordless mechanisms are configured, the secret credential overrides the passwordless identity setting.
+
+## Checklist for new Components
+
+New Components MUST have:
```
Comment (davidfowl) [inline src/Components/README.md:169]: Do we want to say something about what we don't want? Like the fact we try to *avoid* adding new APIs on top of the client APIs?

Comment (eerhardt) [reply src/Components/README.md:169]: Probably not in the checklist, but I can add something along those lines above.

File: src/Components/README.md [hunk 2]
```diff
@@ -163,3 +163,22 @@ builder.AddAzureServiceBus(settings =>
...
+* Tests
+    * [ConformanceTests](../../tests/Aspire.Components.Common.Tests/ConformanceTests.cs)
+    * Other unit tests as needed
+* Tracing support
+
+New Components SHOULD have:
+
+* Logging support
+* Metrics support
+* Health Checks
```
Comment (davidfowl) [inline src/Components/README.md:184]: This should be required as well.

Comment (davidfowl) [reply src/Components/README.md:184]: 😢

**Review by davidfowl — APPROVED** [review]

File: src/Components/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ Aspire components are classic .NET NuGet packages which are designed as highly u
 
 - Each component should provide it's own public and `sealed` `Settings` type.
   > [!NOTE]
-  > This type does not use the name `Options` because it is not an `IOptions`. `IOptions` objects can be configured through dependency injection. These settings needs to be read before the DI container is built, so they can't be `IOptions`.
+  > This type does not use the name `Options` because it is not an `IOptions`. `IOptions` objects can be configured through dependency injection. These settings need to be read before the DI container is built, so they can't be `IOptions`.
```
Comment (sebastienros) [inline src/Components/README.md:15]: Can I suggest to add a link to https://learn.microsoft.com/dotnet/core/extensions/options somehow?

File: src/Components/README.md [hunk 4]
```diff
@@ -163,3 +163,22 @@ builder.AddAzureServiceBus(settings =>
...
 - If both secret and passwordless mechanisms are configured, the secret credential overrides the passwordless identity setting.
+
+## Checklist for new Components
+
+New Components MUST have:
+
+* README.md
+* ConfigurationSchema.json file
+* Public APIs
+    * Extension method off IHostApplicationBuilder and Settings object
```
Comment (sebastienros) [inline src/Components/README.md:174]:
```suggestion
    * Extension method off `IHostApplicationBuilder` and `Settings` object
```

File: src/Components/README.md [hunk 5]
```diff
@@ -163,3 +163,22 @@ builder.AddAzureServiceBus(settings =>
...
+* Tests
+    * [ConformanceTests](../../tests/Aspire.Components.Common.Tests/ConformanceTests.cs)
+    * Other unit tests as needed
+* Tracing support
+
+New Components SHOULD have:
```
Comment (sebastienros) [inline src/Components/README.md:180]:
```suggestion
New components SHOULD have:
```

Same

File: src/Components/README.md (hunk 2 above)
Comment (sebastienros) [inline src/Components/README.md:184]:
```suggestion
* Health checks
```

Comment (eerhardt) [reply src/Components/README.md]: I added a line that says components need to have justification when they don't include these.

**Pull request merged**